### PR TITLE
Add the active and passive oddball task

### PR DESCRIPTION
**Pull request by mscheltienne**

### CODE REVIEW

File: eeg_flow/oddball/oddball.py [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+_TRIGGERS: Dict[str, int] = {
+    "standard": 1,
+    "target": 2,
+    "novel": 3,
+}
+_BACKGROUND_COLOR = (0, 0, 0)  # (r, g, b) between -1 and 1
+_CROSS_WIDTH: int = 20  # pixels
+_CROSS_LENGTH: int = 100  # pixels
+_CROSS_COLOR: str = "black"
+_CROSS_FLICKERING_COLOR: str = "white"
```
Comment (mscheltienne) [inline eeg_flow/oddball/oddball.py]: Task variables to define the cross size, the colors, and the flickering duration.

**Comment by mscheltienne** [comment]
@ArthurNguyen Hey!
This PR adds the oddball task, active and passive. The passive version has a fixation cross with one of the 4 arms changing colors for `_DURATION_FLICKERING` seconds. The arm selected is random, and flickering occurs randomly between 0.3 to 0.6 seconds after the last audio stimulus.

It would be nice if you could test this, it should work as is on any computer. In a fresh python environment, you should:

- Install psychopy: `python -m pip install --progress-bar off psychopy psychtoolbox`
- In GitHub desktop, clone my repository (likely done already) and change the branch to this PR. Click on "Current branch", then change the tab from "Branches" to "Pull requests" and select #23
- Install the package 'eeg_flow' in that fresh env: `python -m pip install -e path/to/eeg_flow`

In that same environment, you can then start the task with the command `eeg_flow-oddball 0a --passive --dev` (Change 0a with another code if you want). 

I will try to test today or tomorrow that the timing are correct on the stim PC you use with an ANT amplifier, to confirm. Then it's just a matter of setting the background color, the cross color(s), the cross size, ... as you desire and to run once each of the trial-list to confirm that it works correctly.

**Comment by ArthurNguyen** [comment]
IMG_5515

hello, many thanks Mathieu!

Currently testing it, first thing I'm noticing is that the right arm appears to be longer than the 3 others. Is that something on your side as well?

And this is a detail for the final step: could we set the "passive" arg to always true? (at the risk of naming the file oddball_passive.py

This is to avoid misunderstanding as technically, all four blocks are in the "passive condition", but only the oddball only (0a 0b) requires the arm flickering, while oddball+game (A and B) do not

(+ when loading one triallist or another, editing the command at the end will be quicker than having to move back before the --passive)

Rather than passive vs active, the difference is actually whether the arm is flickering or not

So we would have a oddball_passive.py, that will have flickers automatically for 0a and 0b, and none for A and B.

And whenever we are back to the initial protocole with no flickers throughout, we'll use a default oddball.py?

Thanks!

**Comment by mscheltienne** [comment]
@ArthurNguyen 

> Currently testing it, first thing I'm noticing is that the right arm appears to be longer than the 3 others. Is that something on your side as well?

Obviously not.. can you give me the values you have set in the task variables, especially the cross length and width?

For the rest, let's discuss it and I'll rewrite it in a way that suits you.

File: eeg_flow/oddball/oddball.py [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+            [-width[0] / 2, -width[1] / 2],  # center-left-bottom
+            [-width[0] / 2, -width[1] / 2 - length[1]],  # bottom-left
+            [width[0] / 2, -width[1] / 2 - length[1]],  # bottom-right
+            [width[0] / 2, -width[1] / 2],  # center-right-bottom
+            [width[0] + length[0], -width[1] / 2],  # right-bottom
+            [width[0] + length[0], width[1] / 2],  # right-up
+            [width[0] / 2, width[1] / 2],  # center-right-up
+            [width[0] / 2, width[1] / 2 + length[1]],  # top-right
+        ]
+    )
```
Comment (mscheltienne) [inline eeg_flow/oddball/oddball.py]: Definition of the cross vertices, maybe I made a mistake using the wrong indices [1] vs [0] for the right arm.

**Comment by mscheltienne** [comment]
@ArthurNguyen Found it, I forgot to divide the width on the X-axis by 2 for the points 8 and 9 (tip of the right arm).

**Pull request merged**